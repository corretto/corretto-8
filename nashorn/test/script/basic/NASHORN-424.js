/*
 * Copyright (c) 2010, 2013, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

/*
 * NASHORN-424: Sparse array cause OutOfMemory exception.
 *
 * @test
 * @run
 */

if (typeof print === 'undefined') {
    print = console.log;
}

var l = -1
  , LEVEL = { silly   : l++
            , verbose : l++
            , info    : l++
            , "http"  : l++
            , WARN    : l++
            , "ERR!"  : l++
            , ERROR   : "ERR!"
            , ERR     : "ERR!"
            , win     : 0x15AAC5
            , paused  : 0x19790701
            , silent  : 0xDECAFBAD
            }

Object.keys(LEVEL).forEach(function (l) {
  if (typeof LEVEL[l] === "string") LEVEL[l] = LEVEL[LEVEL[l]]
  else LEVEL[LEVEL[l]] = l
  LEVEL[l.toLowerCase()] = LEVEL[l]
  if (l === "silent" || l === "paused") return
  log[l] = log[l.toLowerCase()] =
    function (msg, pref, cb) { return log(msg, pref, l, cb) }
})

function log(msg, pref, level, cb) {
    print("[" +level + "] " + msg);
}

Object.keys(LEVEL).forEach(function(l) {
    log("has value " + LEVEL[l], null, l, null);
});



var ar = [ "Hello", "World", 0xDECAFBAD ]

Object.keys(ar).forEach(function(e) {
    print("ar[" + e + "] = " + ar[e]);
});

ar[34254236] = 17;
ar[-1] = "boom";
ar[0xDECAFBAD] = "ka-boom";

ar[ar[2]] = "bye";

Object.keys(ar).forEach(function(e) {
    print("ar[" + e + "] = " + ar[e]);
});

