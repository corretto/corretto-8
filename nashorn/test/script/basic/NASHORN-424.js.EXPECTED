[0] has value verbose
[1] has value info
[2] has value http
[3] has value WARN
[4] has value ERR!
[1419973] has value win
[427362049] has value paused
[3737844653] has value silent
[silly] has value -1
[verbose] has value 0
[info] has value 1
[http] has value 2
[WARN] has value 3
[ERR!] has value 4
[ERROR] has value 4
[ERR] has value 4
[win] has value 1419973
[paused] has value 427362049
[silent] has value 3737844653
[-1] has value silly
[warn] has value 3
[err!] has value 4
[error] has value 4
[err] has value 4
ar[0] = Hello
ar[1] = World
ar[2] = 3737844653
ar[0] = Hello
ar[1] = World
ar[2] = 3737844653
ar[34254236] = 17
ar[3737844653] = bye
ar[-1] = boom
